scalanet/src/io/iohk/scalanet/codec/NodeRecordCode.scala

```diff
@@ -0,0 +1,51 @@
+package io.iohk.scalanet.codec
+
+import io.iohk.scalanet.peergroup.kademlia.{KMessage, KRouter}
+import scodec.Attempt.{Failure, Successful}
+import scodec.{Attempt, Codec, DecodeResult, SizeBound}
+import scodec.bits.BitVector
+
+class NodeRecordCode[A](implicit codec: Codec[A],longCodec:Codec[Long],bitVectorCodec:Codec[BitVector]) extends Codec[KRouter.NodeRecord[A]] {
+  override def decode(bits: BitVector): Attempt[DecodeResult[KRouter.NodeRecord[A]]] = {
+    longCodec.decode(bits) match{
```

Comment:
1. Use for-comp instead pattern matchins on each step
2. You can import `scodec.codecs.Implicits._` into scope to derive codec for `NodeRecord[A]` if there is codec for a in scope